### PR TITLE
feat(AssetsDeployment): Assets deployment

### LINKED ISSUE #10987: Assets deployment
Implement assets deployment.

### PR DESCRIPTION
**Pull request by endulab**
- Add ERC20 contract
- Add decimals field to community_tokens db table
- Adjusting API to handle assets deployment
- Add decimals field to CommunityTokenMetadata

Issue #10987

### CODE REVIEW

**Review by jrainville — APPROVED** [review]
LGTM

File: contracts/assets/CommunityERC20.sol [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    }
+
+    // Events
+
+    // External functions
+
+    function setMaxSupply(uint256 newMaxSupply) external onlyOwner {
+        require(newMaxSupply >= totalSupply(), "MAX_SUPPLY_LOWER_THAN_TOTAL_SUPPLY");
+        maxSupply = newMaxSupply;
+    }
```
Comment (0x-r4bbit) [inline contracts/assets/CommunityERC20.sol:33]: Are we sure we want `maxSupply` to be changable after deployment?

File: contracts/assets/CommunityERC20.sol [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+    /**
+     * @dev Mint tokens for each address in `addresses` each one with
+     * an amount specified in `amounts`.
+     *
+     */
+    function mintTo(address[] memory addresses, uint256[] memory amounts) external onlyOwner {
+        require(addresses.length == amounts.length, "WRONG_LENGTHS");
```
Comment (0x-r4bbit) [inline contracts/assets/CommunityERC20.sol:41]: This error message is not very useful.
I'd recommend using custom errors instead, also saves gas as `require` strings are rather expensive

File: protocol/communities/persistence_test.go [hunk 3]
```diff
@@ -375,7 +375,7 @@ func (s *PersistenceSuite) TestGetCommunityTokens() {
 	s.Require().NoError(err)
 	s.Require().Len(tokens, 0)
 
-	token := CommunityToken{
+	token721 := CommunityToken{
```
Comment (0x-r4bbit) [inline protocol/communities/persistence_test.go]: No biggie, but I think if we **do** make the effort to rename this variable here to make it semantically more meaningful, we should probably call it at least something like `tokenERC721` or so

File: protocol/migrations/sqlite/1687856939_add_community_tokens_decimals.up.sql [hunk 4]
```diff
@@ -0,0 +1 @@
+ALTER TABLE community_tokens ADD COLUMN decimals INT DEFAULT 0;
```
Comment (0x-r4bbit) [inline protocol/migrations/sqlite/1687856939_add_community_tokens_decimals.up.sql]: Should we default to 18 instead?

File: services/collectibles/api.go [hunk 5]
```diff
@@ -109,12 +111,43 @@ func (api *API) Deploy(ctx context.Context, chainID uint64, deploymentParameters
...
+
 // Returns gas units + 10%
 func (api *API) DeployCollectiblesEstimate(ctx context.Context) (uint64, error) {
 	gasAmount := uint64(1960645)
 	return gasAmount + uint64(float32(gasAmount)*0.1), nil
 }
 
+// Returns gas units + 10%
+func (api *API) DeployAssetsEstimate(ctx context.Context) (uint64, error) {
+	gasAmount := uint64(957483)
```
Comment (0x-r4bbit) [inline services/collectibles/api.go:147]: Where does this number come from?
Can we be sure it's constant across different chains?

File: contracts/assets/CommunityERC20.sol (hunk 1 above)
Comment (endulab) [reply contracts/assets/CommunityERC20.sol:33]: Yes, This is our way of burning functionality. The same idea is implemented in ERC721.
We do not have any free tokens in a bank. We just decrease maxSupply, so less tokens can be distributed.
FYI, I did not implement the smart contract. Andrea did.

File: contracts/assets/CommunityERC20.sol (hunk 2 above)
Comment (endulab) [reply contracts/assets/CommunityERC20.sol:41]: FYI, @gravityblast

File: contracts/assets/CommunityERC20.sol (hunk 1 above)
Comment (0x-r4bbit) [reply contracts/assets/CommunityERC20.sol:33]: Ah right, the burn. Gotcha, thanks for reminding

File: services/collectibles/api.go (hunk 5 above)
Comment (endulab) [reply services/collectibles/api.go:147]: It is computed by hardhat deployment testing script.
Yes, it is the same on all chains.

File: protocol/communities/persistence_test.go (hunk 3 above)
Comment (endulab) [reply protocol/communities/persistence_test.go]: Done

File: protocol/migrations/sqlite/1687856939_add_community_tokens_decimals.up.sql (hunk 4 above)
Comment (endulab) [reply protocol/migrations/sqlite/1687856939_add_community_tokens_decimals.up.sql]: Done

File: services/collectibles/api.go (hunk 5 above)
Comment (0x-r4bbit) [reply services/collectibles/api.go:147]: Okay, I think there's a tiny percentage of a chance that L2 clients can implement different gas costs for opcodes at which point this function could estimate incorrectly.

But that's an optimization we can handle if we run into that issue.

**Review by 0x-r4bbit — APPROVED** [review]

**Pull request merged**